### PR TITLE
Day16

### PR DESCRIPTION
**Pull request by Pilchie**

### CODE REVIEW

**Comment by Pilchie** [comment]
@anurse

File: 2020/16/ticket/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+        }
+    }
+
+    fn in_ranges(v: usize, r1: &Range, r2: &Range) -> bool {
+        v >= r1.min && v <= r1.max || v >= r2.min && v <= r2.max
+    }
+}
+
+pub struct ConditionSet {
+    fields: HashMap<String, (Range, Range)>,
```
Comment (anurse) [inline 2020/16/ticket/src/main.rs:46]: I wish Go had tuples instead of just janky "multiple return values"

File: 2020/16/ticket/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+            if line == "" { 
+                break;
+            }
+            let (field, ranges) = split_once(line, ": ");
+            let (first, second) = split_once(ranges, " or ");
+            let (min1, max1) = split_once(first, "-");
+            let (min2, max2) = split_once(second, "-");
+            fields.insert(
+                String::from(field),
+                (Range::new(min1, max1), Range::new(min2, max2)),
```
Comment (anurse) [inline 2020/16/ticket/src/main.rs:62]:
```suggestion
                (min1..=max1, min2..=max2),
```

if you use `RangeInclusive` ;)

**Review by anurse — COMMENTED** [review]
I sat for way too long with a broken Part 2 thinking my input was bugged before I realized I had forgotten to exclude the invalid tickets from Part 1 🤦🏻

File: 2020/16/ticket/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+    }
+
+    Ok(())
+}
+
+#[derive(Debug)]
+pub struct Range {
+    min: usize,
+    max: usize,
+}
```
[the remark this answers is not in the document]
Comment (Pilchie) [reply 2020/16/ticket/src/main.rs:30]: Nice. I should look around the standard libraries more.

File: 2020/16/ticket/src/main.rs (hunk 1 above)
Comment (Pilchie) [reply 2020/16/ticket/src/main.rs:46]: I've been missing C#'s named tuple elements though.

**Pull request merged**